### PR TITLE
Track PR state for prow-based Sippy

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-485](https://issues.redhat.com//browse/TRT-485)

Adds new repository pages that show GitHub repos and the worst performing jobs.

image

image

---
Todo's before merging

- [x] Get a github token available on the DPCR cluster

### CODE REVIEW

File: pkg/db/query/repository_queries.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		Joins("INNER JOIN prow_job_run_prow_pull_requests ON prow_job_run_prow_pull_requests.prow_pull_request_id = prow_pull_requests.id").
+		Joins("INNER JOIN prow_job_runs on prow_job_run_prow_pull_requests.prow_job_run_id = prow_job_runs.id").
+		Joins("INNER JOIN prow_jobs on prow_job_runs.prow_job_id = prow_jobs.id").
+		Joins("LEFT JOIN (?) premerge_failures ON premerge_failures.prow_job_ID = prow_jobs.id", averageByJob).
+		Where("prow_jobs.release = ?", release).
+		Group("prow_pull_requests.org, prow_pull_requests.repo").
+		Select("ROW_NUMBER() OVER() as id, prow_pull_requests.org, prow_pull_requests.repo, coalesce(max(average_premerge_job_failures), 0) as premerge_job_failures, count(distinct(prow_jobs.id)) as job_count")
+
+	results := make([]api.Repository, 0)
+	q, err := filter.FilterableDBResult(dbc.DB.Table("(?) as repos", repos), filterOpts, api.Repository{})
```
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/db/query/repository_queries.go:26]: A trick to make filtering work, `SELECT * FROM (subquery)` makes it so the filtering layer can append WHERE clauses arbitrarily on the api-named fields.

**Comment by stbenjam** [comment]
/hold

One more lint thing to fix, and need to get the token in place

**Comment by stbenjam** [comment]
/hold cancel

@dgoodwin This is ready for another look

File: pkg/prowloader/prow.go [hunk 2]
```diff
@@ -136,6 +145,30 @@ func (pl *ProwLoader) LoadProwJobsToDB() error {
...
 }
 
+func (pl *ProwLoader) syncPRStatus() error {
+	if pl.githubClient == nil {
+		log.Infof("No GitHub client, skipping PR sync")
+		return nil
+	}
+
+	pulls := make([]models.ProwPullRequest, 0)
+	pl.dbc.DB.Table("prow_pull_requests").Where("merged_at IS NULL").Scan(&pulls)
```
Comment (dgoodwin) [inline pkg/prowloader/prow.go:155]: Looks like a missing error check here.

File: pkg/prowloader/prow.go [hunk 3]
```diff
@@ -136,6 +145,30 @@ func (pl *ProwLoader) LoadProwJobsToDB() error {
...
+	for _, pr := range pulls {
+		mergedAt, err := pl.githubClient.GetPRMerged(pr.Org, pr.Repo, pr.Number, pr.SHA)
+		if err != nil {
+			log.WithError(err).Warningf("could not fetch pull request status from GitHub; org=%q repo=%q number=%q sha=%q", pr.Org, pr.Repo, pr.Number, pr.SHA)
+			return err
+		}
+		pr.MergedAt = mergedAt
```
Comment (dgoodwin) [inline pkg/prowloader/prow.go:162]: Should we check if it changed here before doing the Save to limit writes?

**Review by dgoodwin — COMMENTED** [review]
Just a couple small ones, otherwise looks good.

File: pkg/prowloader/prow.go (hunk 2 above)
Comment (stbenjam) [reply pkg/prowloader/prow.go]: Thanks, fixed

File: pkg/prowloader/prow.go (hunk 3 above)
Comment (stbenjam) [reply pkg/prowloader/prow.go]: Good call, fixed

**Comment by stbenjam** [comment]
@dgoodwin Addressed your feedback in [7ed9f70](https://github.com/openshift/sippy/pull/484/commits/7ed9f709ab1104094ac14dd4e44a37173c13eaee)

**Pull request merged**